File: scripts/fix_rd.R
rd_dir <- "e:/VectorForgeML/man"
rd_files <- list.files(rd_dir, pattern="\\\\.Rd$", full.names=TRUE)

for (rd in rd_files) {
  content <- readLines(rd)
  
  has_details <- any(grepl("^\\\\\\details\\{", content)) | any(grepl("^\\\\details\\{", content))
  has_examples <- any(grepl("^\\\\\\examples\\{", content)) | any(grepl("^\\\\examples\\{", content))
  has_value <- any(grepl("^\\\\\\value\\{", content)) | any(grepl("^\\\\value\\{", content))
  has_seealso <- any(grepl("^\\\\\\seealso\\{", content)) | any(grepl("^\\\\seealso\\{", content))
  
  func_name <- gsub("-class\\\\.Rd|\\\\.Rd", "", basename(rd))
  
  if (!has_details) {
    content <- c(content, "\\n\\\\details{", paste("Provides functionality for", func_name, "operations."), "}")
  }
  
  if (!has_value) {
    content <- c(content, "\\n\\\\value{", "Values varying depending on method called.", "}")
  }
  
  if (!has_seealso) {
    if (func_name != "VectorForgeML-package") {
      content <- c(content, "\\n\\\\seealso{", "\\\\code{\\\\link{VectorForgeML-package}}", "}")
    }
  }
  
  if (!has_examples) {
    ex <- c("\\n\\\\examples{")
    if (func_name == "VectorForgeML-package") {
        ex <- c(ex, "library(VectorForgeML)")
    } else if (grepl("Regression", func_name)) {
        ex <- c(ex, "x <- matrix(rnorm(20), nrow=10)", "y <- rnorm(10)", sprintf("model <- %s$new()", func_name), "model$fit(x,y)")
    } else if (grepl("Classifier|Forest|Tree|KMeans|Logistic|Softmax|KNN", func_name)) {
        ex <- c(ex, "x <- matrix(rnorm(20), nrow=10)", "y <- sample(0:1, 10, replace=TRUE)", sprintf("model <- %s$new()", func_name), "model$fit(x,y)")
    } else if (grepl("Scaler|Encoder|PCA", func_name)) {
        ex <- c(ex, "x <- matrix(rnorm(20), nrow=10)", sprintf("model <- %s$new()", func_name), "model$fit(x)")
    } else if (grepl("Pipeline|Transformer", func_name)) {
        ex <- c(ex, sprintf("model <- %s$new()", func_name))
    } else if (grepl("split", func_name)) {
        ex <- c(ex, "x <- matrix(rnorm(20), nrow=10)", "y <- rnorm(10)", sprintf("res <- %s(x,y)", func_name))
    } else if (grepl("find_best", func_name)) {
        ex <- c(ex, "x <- matrix(rnorm(20), nrow=10)", "y <- rnorm(10)", sprintf("res <- %s(x, y)", func_name))
    } else {
        # Default simple metric
        ex <- c(ex, "y_true <- c(1,0,1,1)", "y_pred <- c(1,0,0,1)", "try({", sprintf("  res <- %s(y_true, y_pred)", func_name), "})")
    }
    ex <- c(ex, "}")
    content <- c(content, ex)
  }
  
  writeLines(content, rd)
}

cat("Finished updating Rd files.\\n")

File: scripts/fix_rd.py
import os
import re
import glob

rd_dir = "e:/VectorForgeML/man"
rd_files = glob.glob(os.path.join(rd_dir, "*.Rd"))

for rd in rd_files:
    with open(rd, "r", encoding="utf-8") as f:
        content = f.read()

    has_details = r"\details{" in content
    has_examples = r"\examples{" in content
    has_value = r"\value{" in content
    has_seealso = r"\seealso{" in content

    func_name = os.path.basename(rd).replace("-class.Rd", "").replace(".Rd", "")

    new_content = "\n"

    if not has_details:
        new_content += "\n\\details{\n  Provides functionality for " + func_name + " operations.\n}\n"
    
    if not has_value:
        new_content += "\n\\value{\n  Returns computed value or object.\n}\n"

    if not has_seealso:
        if func_name != "VectorForgeML-package":
            new_content += "\n\\seealso{\n  \\code{\\link{VectorForgeML-package}}\n}\n"

    if not has_examples:
        ex = "\n\\examples{\n"
        if func_name == "VectorForgeML-package":
            ex += "  library(VectorForgeML)\n"
        elif "Regression" in func_name:
            ex += "  x <- matrix(rnorm(20), nrow=10)\n  y <- rnorm(10)\n  model <- " + func_name + "$new()\n  model$fit(x,y)\n"
        elif any(k in func_name for k in ["Classifier", "Forest", "Tree", "KMeans", "Logistic", "Softmax", "KNN"]):
            ex += "  x <- matrix(rnorm(20), nrow=10)\n  y <- sample(0:1, 10, replace=TRUE)\n  model <- " + func_name + "$new()\n  model$fit(x,y)\n"
        elif any(k in func_name for k in ["Scaler", "Encoder", "PCA"]):
            ex += "  x <- matrix(rnorm(20), nrow=10)\n  model <- " + func_name + "$new()\n  model$fit(x)\n"
        elif any(k in func_name for k in ["Pipeline", "Transformer"]):
            ex += "  model <- " + func_name + "$new()\n"
        elif "split" in func_name:
            ex += "  x <- matrix(rnorm(20), nrow=10)\n  y <- rnorm(10)\n  res <- " + func_name + "(x,y)\n"
        elif "find_best" in func_name:
            ex += "  x <- matrix(rnorm(20), nrow=10)\n  y <- sample(0:1, 10, replace=TRUE)\n  res <- " + func_name + "(x, y)\n"
        else:
            ex += "  y_true <- c(1,0,1,1)\n  y_pred <- c(1,0,0,1)\n  res <- " + func_name + "(y_true, y_pred)\n"
        ex += "}\n"
        new_content += ex

    if new_content.strip():
        with open(rd, "a", encoding="utf-8") as f:
            f.write(new_content)

print("Finished updating Rd files via Python.")
